src/briefcase/integrations/android_sdk.py

```diff
@@ -74,11 +95,11 @@ def cmdline_tools_path(self) -> Path:
     @property
     def cmdline_tools_version(self) -> str:
         # This is the version of the Android SDK Command-line tools that
-        # are current as of May 2022. These tools can generally self-update,
+        # are current as of July 2023. These tools can generally self-update,
         # so using a fixed download URL isn't a problem.
         # However, if/when this version number is changed, ensure that the
         # checks done during verification include any required upgrade steps.
-        return "8092744"
+        return "9477386"
```

Comment:
First, thanks for bringing it up; I read the first part of this paragraph saying "these tools can generally self-update" and apparently didn't really digest the rest of it.

Second, Google's [release notes](https://developer.android.com/tools/releases/cmdline-tools) for the Command-line Tools is great 😄.

So, I'm quite uncertain what has changed between download `8092744` and `9477386`. I can say that `sdkmanager` reports versions `6.0` and `9.0`, respectively. A lot of the files have identical hashes...but many of the JARs do not.

Nonetheless, I believe Briefcase just uses `sdkmanager` and `avdmanager` from `cmdline-tools`. These tools do not appear to have changed in perceivable ways for the tasks we use them for.

@mhsmith, do you have insight in to how to know if it is safe to start using a new version of Command-line Tools? It definitely feels like "try it out and fix what breaks".

I can revert the change....but I feel like we should have a methodology to bump this version.

> are there any migration steps we should run for existing users?

I imagine this and the note in the code, "ensure that the checks done during verification include any required upgrade steps", derive from Android's switch a few years back from SDK Tools to Command-line tools....which was very disruptive.

Reply:
FWIW, I ran through the crossover a user might experience.

I started with a clean slate via:
```console
❯ rm -rf ~/.gradle ~/.android/ ~/.cache/briefcase/tools/android_sdk ~/.cache/briefcase/tools/java17/ ~/.cookiecutter*
```
And then ran `briefcase build android` using Briefcase from `main`. This installed using cmd-line tools 8092744.

I, then, got rid of the Android SDK to start with a new one: `rm -rf ./.cache/briefcase/tools/android_sdk`

Using Briefcase from this PR, I ran `briefcase run android -ur`. It installed cmd-line tools 9477386, rebuilt the ADB, and was able to reuse the AVD I created with the first run. It needed to download the system image again...but I think that's because they live in the Android SDK itself in the Briefcase data directory.

Finally, running `briefcase upgrade android_sdk` worked as well...although, it didn't need to update anything.

This all worked for me on Windows, as well.

Reply:
> @mhsmith, do you have insight in to how to know if it is safe to start using a new version of Command-line Tools? It definitely feels like "try it out and fix what breaks".

Yes, as long as it's a stable version it should be fairly safe to update. So I think this manual testing is enough.

Reply:
Thinking more about this, it'll probably be best to force an upgrade of Command-line Tools when its version is bumped. Otherwise, users could possibly remain on an older version of these tools indefinitely.

Reply:
For previous notes on upgrading the command-line tools, see https://github.com/beeware/briefcase/pull/766#issuecomment-1160947331.

Reply:
Oh, awesome; thanks!

That looks pretty much like what I'm implementing right now; instead of installing to `latest`, Briefcase would install more like how Android Studio does and create version-specific installs of Command-line Tools.

Reply:
- I moved this upgrade to to https://github.com/beeware/briefcase/pull/1397